### ISSUE TITLE
Checkstyle shows diagnostics in the wrong line

### ISSUE THREAD

**Issue by ismaelpadilla** [mfussenegger/nvim-lint#249]
It looks like if Checkstyle reports an error in line L, column C, the diagnostic is shown in line C (instead of line L).

Here are a few screenshots:
- Error in line 6, column 7, is shown in line 7:
![Erron in 6:7](https://user-images.githubusercontent.com/7772501/191270424-801b9163-dd88-4eed-a8c2-b63917cf69f0.png)

- Error in line 6, column 4, is shown in line 4:
![Erron in 6:4](https://user-images.githubusercontent.com/7772501/191270435-adf5380a-7a09-4466-a628-1cfec72783ec.png)

- Error in line 6, column 3, is shown in line 3:
![Erron in 6:3](https://user-images.githubusercontent.com/7772501/191271362-49731a09-eeb8-4b8b-9361-efb53491f511.png)

**Comment by mfussenegger** [comment]
For me the errors are reported in the correct lines:


```java


public class Main {
   public static void main(String[] args) throws Exception {
    System.out.println();
  }
}
```


```
Foo.java|1-2 col 1 warning| Foo.java is a non-project file, only syntax errors are reported
Foo.java|3 col 1 warning| The name of the outer type and the file do not match. [OuterTypeFilename]
Foo.java|3 col 1 warning| Missing a Javadoc comment. [MissingJavadocType]
Foo.java|4 col 4 warning| 'method def modifier' has incorrect indentation level 3, expected level should be 2. [Indentation]
```

![image](https://user-images.githubusercontent.com/38700/191365491-1719ee6a-5228-4b7d-b8ee-17d5cd865812.png)

**Comment by ismaelpadilla** [comment]
You're right, I didn't test this thoroughly enough before opening the issue. Everything works fine for me when using the default rules, but I get the error when using custom rules.

This is the xml that I am using, which I am setting with:
```lua
require('lint').linters.checkstyle.config_file = '/some/path/checkstyle_rules.xml'
```

My rules file:
<details>
    <summary>checkstyle_rules.xml</summary>

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="localeCountry" value="US"/>
    <property name="localeLanguage" value="en"/>
    <property name="charset" value="UTF-8"/>
    <property name="fileExtensions" value="java, properties"/>
    <!-- <module name="SuppressionFilter"> -->
    <!--     <property name="file" value="config/checkstyle_suppressions.xml"/> -->
    <!--     <property name="optional" value="true"/> -->
    <!-- </module> -->
    <module name="FileTabCharacter">
        <property name="eachLine" value="true"/>
    </module>
    <module name="LineLength">
        <property name="fileExtensions" value="java"/>
        <property name="max" value="140"/>
        <property name="ignorePattern" value='^package.*|^import.*|a href|href|http://|https://|ftp://|.*i18nService.tr.*\",$'/>
    </module>
    <module name="NewlineAtEndOfFile"/>
    <module name="OrderedProperties"/>
    <module name="UniqueProperties"/>
    <module name="TreeWalker">
        <module name="AnnotationLocation">
            <property name="id" value="AnnotationLocationMostCases"/>
            <property name="tokens"
                      value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF,
                      RECORD_DEF, COMPACT_CTOR_DEF"/>
        </module>
        <module name="AnnotationLocation">
            <property name="id" value="AnnotationLocationVariables"/>
            <property name="tokens" value="VARIABLE_DEF"/>
            <property name="allowSamelineMultipleAnnotations" value="true"/>
        </module>
        <module name="AnnotationUseStyle">
            <property name="elementStyle" value="compact"/>
            <property name="trailingArrayComma" value="never"/>
            <property name="closingParens" value="never"/>
        </module>
        <module name="ArrayTrailingComma">
            <property name="alwaysDemandTrailingComma" value="false"/>
        </module>
        <module name="ArrayTypeStyle">
            <property name="javaStyle" value="true"/>
        </module>
        <module name="AtclauseOrder">
            <property name="tagOrder" value="@param, @return, @throws, @deprecated"/>
            <property name="target"
                      value="CLASS_DEF, INTERFACE_DEF, ENUM_DEF, METHOD_DEF, CTOR_DEF, VARIABLE_DEF"/>
        </module>
        <module name="AvoidDoubleBraceInitialization"/>
<!--        <module name="AvoidEscapedUnicodeCharacters"/>-->
        <module name="AvoidNestedBlocks">
            <property name="allowInSwitchCase" value="false"/>
        </module>
        <module name="AvoidNoArgumentSuperConstructorCall"/>
        <module name="AvoidStarImport"/>
        <module name="UnusedImports"/>
        <module name="CatchParameterName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Catch parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="ClassTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Class type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="CommentsIndentation"/>
        <module name="ConstantName">
            <property name="format" value="^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"/>
        </module>
        <module name="CustomImportOrder">
            <property name="customImportOrderRules" value="STATIC###THIRD_PARTY_PACKAGE"/>
            <property name="sortImportsInGroupAlphabetically" value="true"/>
            <property name="separateLineBetweenGroups" value="true"/>
        </module>
        <module name="DeclarationOrder"/>
        <module name="DefaultComesLast"/>
        <module name="EmptyBlock"/>
        <module name="EmptyCatchBlock">
            <property name="commentFormat" value=".*"/>
            <property name="exceptionVariableName" value=".*"/>
        </module>
        <module name="EmptyForInitializerPad"/>
        <module name="EmptyForIteratorPad"/>
        <module name="EmptyLineSeparator">
            <property name="tokens"
                      value="PACKAGE_DEF, IMPORT, STATIC_IMPORT, CLASS_DEF, INTERFACE_DEF, ENUM_DEF,
                            STATIC_INIT, INSTANCE_INIT, METHOD_DEF, CTOR_DEF, VARIABLE_DEF, RECORD_DEF,
                            COMPACT_CTOR_DEF"/>
            <property name="allowNoEmptyLineBetweenFields" value="true"/>
        </module>
        <module name="EmptyStatement"/>
        <module name="ExplicitInitialization"/>
        <module name="GenericWhitespace">
            <message key="ws.followed"
                     value="GenericWhitespace ''{0}'' is followed by whitespace."/>
            <message key="ws.preceded"
                     value="GenericWhitespace ''{0}'' is preceded with whitespace."/>
            <message key="ws.illegalFollow"
                     value="GenericWhitespace ''{0}'' should be followed by whitespace."/>
            <message key="ws.notPreceded"
                     value="GenericWhitespace ''{0}'' is not preceded with whitespace."/>
        </module>
        <module name="HiddenField">
            <property name="ignoreSetter" value="true"/>
            <property name="ignoreConstructorParameter" value="true"/>
        </module>
        <module name="IllegalIdentifierName">
            <property name="format" value="(?i)^(?!(record|yield|var|permits|sealed|_)$).+$"/>
        </module>
        <module name="IllegalThrows"/>
        <module name="Indentation">
            <property name="basicOffset" value="2"/>
            <property name="braceAdjustment" value="2"/>
            <property name="caseIndent" value="2"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="2"/>
        </module>
        <module name="InnerAssignment"/>
        <module name="InterfaceTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Interface type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MissingDeprecated"/>
        <module name="MissingOverride"/>
        <module name="LeftCurly"/>
        <module name="NeedBraces"/>
        <module name="RightCurly"/>
<!--        <module name="magicnumber">-->
<!--            <property name="ignorenumbers" value="-1,0,1"/>-->
<!--            <property name="ignorefielddeclaration" value="true"/>-->
<!--            <property name="ignoreannotation" value="true"/>-->
<!--        </module>-->
        <module name="MissingSwitchDefault"/>
        <module name="ModifiedControlVariable"/>
<!--        <module name="MultipleStringLiterals"/>-->
        <module name="MultipleVariableDeclarations"/>
        <module name="OneStatementPerLine"/>
        <module name="OverloadMethodsDeclarationOrder"/>
        <module name="PackageDeclaration"/>
        <module name="ParameterAssignment"/>
        <module name="SimplifyBooleanExpression"/>
        <module name="SimplifyBooleanReturn"/>
        <module name="StringLiteralEquality"/>
        <module name="UnnecessarySemicolonAfterTypeMemberDeclaration"/>
        <module name="UnnecessarySemicolonInTryWithResources"/>
        <module name="InvalidJavadocPosition"/>
        <module name="JavadocBlockTagLocation"/>
        <module name="JavadocContentLocationCheck"/>
<!--        <module name="JavadocMethod">-->
<!--            <property name="accessModifiers" value="public"/>-->
<!--            <property name="validateThrows" value="true"/>-->
<!--            <property name="allowedAnnotations" value="Override, Test"/>-->
<!--        </module>-->
        <module name="JavadocMissingLeadingAsterisk"/>
        <module name="JavadocMissingWhitespaceAfterAsterisk"/>
        <module name="JavadocParagraph"/>
        <module name="JavadocStyle">
            <property name="scope" value="public"/>
        </module>
        <module name="JavadocTagContinuationIndentation"/>
        <module name="JavadocType">
            <property name="scope" value="public"/>
        </module>
<!--        <module name="MissingJavadocMethod">-->
<!--            <property name="scope" value="public"/>-->
<!--            <property name="allowedAnnotations" value="Override, Test"/>-->
<!--        </module>-->
<!--        <module name="MissingJavadocType">-->
<!--            <property name="scope" value="public"/>-->
<!--            <property name="tokens" value="INTERFACE_DEF"/>-->
<!--            <property name="skipAnnotations" value="Generated, SpringBootApplication, Configuration"/>-->
<!--        </module>-->
        <module name="NonEmptyAtclauseDescription"/>
        <module name="SingleLineJavadoc"/>
        <module name="NoCodeInFile"/>
        <module name="OuterTypeFilename"/>
        <module name="TrailingComment"/>
<!--        <module name="UncommentedMain"/>-->
        <module name="UpperEll"/>
        <module name="ModifierOrder"/>
        <module name="RedundantModifier"/>
        <module name="LambdaParameterName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Lambda parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="LocalFinalVariableName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Local final variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="LocalVariableName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Local variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MemberName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Member name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MethodName">
            <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>
            <message key="name.invalidPattern"
                     value="Method name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="MethodTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Method type name ''{0}'' must match pattern ''{1}''."/>
        </module>
<!--        <module name="PackageName">-->
<!--            <property name="format" value="^[a-z]+(\.[a-z][a-z0-9]*)*$"/>-->
<!--            <message key="name.invalidPattern"-->
<!--                     value="Package name ''{0}'' must match pattern ''{1}''."/>-->
<!--        </module>-->
        <module name="ParameterName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Parameter name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="PatternVariableName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Pattern variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="RecordComponentName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Record component name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="RecordTypeParameterName">
            <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
            <message key="name.invalidPattern"
                     value="Record type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="StaticVariableName">
            <property name="format" value="^[a-z]([a-z0-9][a-zA-Z0-9]*)?$"/>
            <message key="name.invalidPattern"
                     value="Static variable name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="TypeName">
            <property name="format" value="^[A-Z][a-zA-Z0-9]*$"/>
            <message key="name.invalidPattern"
                     value="Type name ''{0}'' must match pattern ''{1}''."/>
        </module>
        <module name="SeparatorWrap">
            <property name="tokens" value="DOT"/>
            <property name="option" value="nl"/>
        </module>
        <module name="MethodParamPad"/>
        <module name="NoLineWrap"/>
        <module name="NoWhitespaceAfter"/>
        <module name="NoWhitespaceBefore"/>
        <module name="OperatorWrap"/>
        <module name="ParenPad"/>
        <module name="SingleSpaceSeparator"/>
        <module name="TypecastParenPad"/>
        <module name="WhitespaceAfter"/>
        <module name="WhitespaceAround">
            <property name="allowEmptyConstructors" value="true"/>
            <property name="allowEmptyLambdas" value="true"/>
            <property name="allowEmptyMethods" value="true"/>
            <property name="allowEmptyTypes" value="true"/>
            <property name="allowEmptyLoops" value="true"/>
            <property name="ignoreEnhancedForColon" value="false"/>
            <property name="tokens"
                      value="ASSIGN, BAND, BAND_ASSIGN, BOR, BOR_ASSIGN, BSR, BSR_ASSIGN, BXOR,
                            BXOR_ASSIGN, COLON, DIV, DIV_ASSIGN, DO_WHILE, EQUAL, GE, GT, LAMBDA, LAND,
                            LCURLY, LE, LITERAL_CATCH, LITERAL_DO, LITERAL_ELSE, LITERAL_FINALLY,
                            LITERAL_FOR, LITERAL_IF, LITERAL_RETURN, LITERAL_SWITCH, LITERAL_SYNCHRONIZED,
                            LITERAL_TRY, LITERAL_WHILE, LOR, LT, MINUS, MINUS_ASSIGN, MOD, MOD_ASSIGN,
                            NOT_EQUAL, PLUS, PLUS_ASSIGN, QUESTION, RCURLY, SL, SLIST, SL_ASSIGN, SR,
                            SR_ASSIGN, STAR, STAR_ASSIGN, LITERAL_ASSERT, TYPE_EXTENSION_AND"/>
        </module>
    </module>
</module>
```
</details>

I've also noticed that this issue goes away by modifying the file `lua/lint/linters/checkstyle.lua`, and changing the first line from:
```lua
local format = '[%tRROR] %f:%l: %m, [%tRROR] %f:%l:%c: %m, [%tARN] %f:%l:%c: %m, [%tARN] %f:%l: %m'
```

to:
```lua
local format = '[%tRROR] %f:%l:%c: %m, [%tRROR] %f:%l: %m, [%tARN] %f:%l:%c: %m, [%tARN] %f:%l: %m'
```

This works fine with my custom config and with the default rules. 

I have submitted a PR (#250), but I marked it as a draft because I don't know enough about error formats, diagnostics, or the tools involved to know if this can cause any other issue. 

To me, it maybe looks like an error like this one:
> [ERROR] /Users/ipadilla/Test/Main.java:6:3: 'method def' child has incorrect indentation level 2, expected level should be 4. [Indentation]


Is being matched against the `[%tRROR] %f:%l: %m` error format, which is maybe setting the file to `[ERROR] /Users/ipadilla/Test/Main.java:6` and the line to `3`?

I've also noticed that the error format for markdown is `'%f:%l:%c %m,%f:%l %m'`, which defines the rule with a column number before than the rule without it, similarly to the change I am proposing.

**Comment by ismaelpadilla** [comment]
I modified the checkstyle test to add a case where the current code fails, but my PR doesn't ([see commit](https://github.com/mfussenegger/nvim-lint/pull/250/commits/ad39980784127546a3ca663a3439e8fb5ec9c3c0)):
> [ERROR] src/main/java/com/foo/bar/ApiClient.java:6:3: 'method def' child has incorrect indentation level 2, expected level should be 4. [Indentation]

If you run that test with the current code, you'll see that the column is set as 0 and the line number is set as 5

**Comment by ismaelpadilla** [comment]
I'm closing this since the PR (#250) was merged 😊
